app/views/agents/_form.html.erb

```diff
@@ -85,9 +85,9 @@
               <%= f.label :sources %>
               <span class="glyphicon glyphicon-question-sign hover-help" data-content="This Agent will receive events from the selected Agents."></span>
               <div class="link-region" data-can-receive-events="<%= @agent.can_receive_events? %>">
-                <% eventSources = (current_user.agents - [@agent]).find_all { |a| a.can_create_events? } %>
+                <% eventSources = (current_user.agents + Agent.shared - [@agent]).find_all { |a| a.can_create_events? } %>
                 <%= f.select(:source_ids,
-                             options_for_select(eventSources.map {|s| [s.name, s.id] },
+                             options_for_select(eventSources.map {|s| [s.name, s.id, {'data-username': s.user.username, 'data-shared': s.user != @agent.user}] },
```

Comment:
Ah good point - will switch this to a 2.1 compatible syntax - no sense dropping 2.1 support just for this